cmds/boot/boot/boot.go

```diff
@@ -63,13 +65,31 @@ func updateBootCmdline(cl string) string {
 func main() {
 	flag.Parse()
 
+	if *verbose {
+		block.Debug = log.Printf
+	}
 	blockDevs, err := block.GetBlockDevices()
 	if err != nil {
 		log.Fatal("No available block devices to boot from")
 	}
 
 	// Try to only boot from "good" block devices.
 	blockDevs = blockDevs.FilterZeroSize()
+
+	// Parse and filter blocklist
+	if *blockList != "" {
```

Comment:
so, overall, were this a plan 9 system, the filter would be a regexp, because things are names, not numbers. And you'd get the following benefit: a regexp lets you define match and reject, whereas this only lets you do a reject.

so it would not be a blocklist, it would be a regexp that, again, could let you match and reject.

It is possible that one regexp would not be enough, but still ... this setup seems unnecessarily restrictive. If you look at the output of the pci command, you can imagine filtering on, e.g., anything matching Broadcomm. 

This is basically how it all worked in Plan 9. The tool of choice for filtering sockets, pci devices, and file names was the same tool: grep. It would be nice to maintain that here.

What you're doing here is limiting yourself to vendor:devid, and there are way more choices on which to match/reject than that. For example, I might want only function 0 of a certain dev:vendid (this came up in practice), and this change can't do that. But a regexp can: ".*:...0 8086 0104" for one example, will filter for only function 0 of all devices made by intel that are 0104. And of course rather than ven:dev pairs, we can even do something like this:
.*:...0.*NetXtreme.*BCM77.*
will get function 0 of a much more recognicable name than a vendor/devid.

You might consider doing this with regexps?

Reply:
Maybe I don't understand sysfs well enough. However while I can take a regex for vendor and devid, I'm not entirely sure how I would search for the backing pci bdf in sysfs given a name in /sys/class/block. Do i start parsing the relative directory names and start trying to treat them as a pci bdf? I'm not sure how to extend this in a sane way.

Reply:
I probably did not understand what you are doing here ... oops.